cvat/apps/dataset_manager/bindings.py

```diff
@@ -482,7 +503,339 @@ def match_frame_fuzzy(self, path):
+                if 'track_id' in shape:
+                    if shape['outside']:
+                        continue
+                    exported_shape = self._export_tracked_shape(shape, task.id)
+                else:
+                    exported_shape = self._export_labeled_shape(shape, task.id)
+                get_frame(task.id, shape['frame']).labeled_shapes.append(exported_shape)
+
+            for tag in self._annotation_irs[task.id].tags:
+                get_frame(task.id, tag['frame']).tags.append(self._export_tag(tag, task.id))
+
+        return iter(frames.values())
+
+    @property
+    def shapes(self):
+        for task in self._db_tasks.values():
+            for shape in self._annotation_irs[task.id].shapes:
+                yield self._export_labeled_shape(shape, task.id)
+
+    @property
+    def tracks(self):
+        idx = 0
+        for task in self._db_tasks.values():
+            for track in self._annotation_irs[task.id].tracks:
+                tracked_shapes = TrackManager.get_interpolated_shapes(
+                    track, 0, task.data.size
+                )
+                for tracked_shape in tracked_shapes:
+                    tracked_shape["attributes"] += track["attributes"]
+                    tracked_shape["track_id"] = idx
+                    tracked_shape["group"] = track["group"]
+                    tracked_shape["source"] = track["source"]
+                    tracked_shape["label_id"] = track["label_id"]
+                yield ProjectData.Track(
+                    label=self._get_label_name(track["label_id"]),
+                    group=track["group"],
+                    source=track["source"],
+                    shapes=[self._export_tracked_shape(shape, task.id)
+                        for shape in tracked_shapes],
+                    task_id=task.id
+                )
+                idx+=1
+
+    @property
+    def tags(self):
+        for task in self._db_tasks.values():
+            for tag in self._annotation_irs[task.id].tags:
+                yield self._export_tag(tag, task.id)
+
+    @property
+    def meta(self):
+        return self._meta
+
+    @property
+    def data(self):
+        raise NotImplementedError()
+
+    @property
+    def frame_info(self):
+        return self._frame_info
+
+    @property
+    def frame_step(self):
+        return self._frame_steps
+
+    @property
+    def db_project(self):
+        return self._db_project
+
+    @property
+    def subsets(self) -> List[str]:
+        return self._subsets
+
+    @property
+    def tasks(self):
+        return list(self._db_tasks.values())
+
+    @property
+    def task_data(self):
+        for task_id, task in self._db_tasks.items():
+            yield TaskData(self._annotation_irs[task_id], task, self._host)
+
+    @staticmethod
+    def _get_filename(path):
+        return osp.splitext(path)[0]
+
+
+class CVATDataExtractor(datumaro.SourceExtractor):
+    dm_items: List[datumaro.DatasetItem] = []
```

Comment:
This is antipattern - such declaration will share the value assigned here between instances. To fix, use the attrs library (https://github.com/openvinotoolkit/datumaro/blob/develop/datumaro/components/extractor.py#L31) or assign values in the `__init__` method.